docs/faq.rst

```diff
@@ -1255,6 +1255,43 @@ It may be useful to set ``BORG_RELOCATED_REPO_ACCESS_IS_OK=yes`` to avoid the
 prompts when renaming multiple repositories or in a non-interactive context
 such as a script. See :doc:`deployment` for an example.
 
+The repository quota size is reached, what can I do?
+----------------------------------------------------
+
+The simplest solution is to increase or disable the quota and resume the backup:
+
+::
+
+    borg config /path/to/repo storage_quota 0
+
+If you are bound to the quota, you have to free repository space. The first to
+try is running :ref:`borg_compact` to free unused backup space (see also
+:ref:`separate_compaction`):
+
+::
+
+    borg compact /path/to/repo
+
+If your repository is already compacted, run :ref:`borg_prune` or
+:ref:`borg_delete` to delete archives that you do not need anymore, and then run
+``borg compact`` again.
+
+My backup disk is full, what can I do?
+--------------------------------------
+
+Borg cannot work if you really have zero free space on the backup disk, so the
+first thing you must do is deleting some files to regain free disk space. See
+:ref:`about_free_space` for further details.
+
+Next you should run :ref:`borg_check` to find out if there is any error in your
+backup repository. If something is wrong, try to repair it first before running
+any other Borg command.
+
```

Comment:
Is there a special reason to do that in this situation?

Of course one can do that at any time to check the state of the repo, but I don't see a special need to do this after a disk full problem. borg should be able to roll back any uncommitted segments to return to a clean state.

Reply:
Yes, it should. I would definitely check my repo after disk full, before I use it again for backups.

Reply:
I have added "first thing you must do is deleting some files to regain free disk space" before, to make sure `borg check` works. I cannot test this in a realistic setting (like an encrypted repo on 8 or 16 GB USB stick) at the moment, but it is definitely interesting to see if our theoretical recipes work in reality...

Reply:
I did some disk-full tests yesterday. IMO there are two relevant conditions / cases:
1. disk already full when Borg starts - Borg almost immediately stops working, nothing bad happens
2. disk gets full when Borg saves chunks - a rollback happens and at least the new segment file is deleted, so that the disk is not full anymore

In case 2 my `borg check` didn't find any issues, so the rollback was successfull). Because of this I will change the "full disk faq" to note these two cases and that nothing special must be done regarding Borg.